src/server/notifications/StreamingHTTPChannel2023/StreamingHTTP2023Emitter.ts

```diff
@@ -0,0 +1,40 @@
+import { getLoggerFor } from '../../../logging/LogUtil';
+import { readableToString } from '../../../util/StreamUtil';
+import { StreamingHTTPMap } from './StreamingHTTPMap';
+
+import type { Representation } from '../../../http/representation/Representation';
+import { AsyncHandler } from '../../../util/handlers/AsyncHandler';
+import { NotificationChannel } from '../NotificationChannel';
+
+export interface StreamingHTTPEmitterInput {
+  channel: NotificationChannel;
+  representation: Representation;
+}
+
+/**
+ * Emits notifications on StreamingHTTPChannel2023 streams.
+ * Uses the StreamingHTTPs found in the provided map.
+ * The key should be the identifier of the topic resource.
+ */
+export class StreamingHTTP2023Emitter extends AsyncHandler<StreamingHTTPEmitterInput>  {
+  protected readonly logger = getLoggerFor(this);
+
+  constructor(
+    private readonly streamMap: StreamingHTTPMap
+  ) {
+    super()
+  }
+
+  public async handle({ channel, representation }: StreamingHTTPEmitterInput): Promise<void> {
+    // Called as a NotificationEmitter: emit the notification
+    const streams = this.streamMap.get(channel.topic);
+    if (streams) {
+      const data = await readableToString(representation.data);
+      for (const stream of streams) {
+        stream.write(data);
+      }
```

Comment:
Not that relevant as this is always going to be a small string, but you could also pipe the data stream directly into the stored streams.